### ISSUE TITLE
Enforce specifying array value type

### ISSUE THREAD

**Issue by rieschl** [vimeo/psalm#10287]
When having an `array` parameter or return type, psalm does not check if there is a type of the values specified in the docblock.
It would be nice to enforce specifying a type (or multiple types) of the array values.

For example:
```php
/**
 * @param array $data
 * @return array
 */
function arrayShapes(array $data): array {
    if (count($data) > 1) {
        return ['bar'];
    }
    return [2];
}
```
([code snippet in the playground](https://psalm.dev/r/7d42144f5e))
This should emit errors that no array type is specified, just like PHPStan does ([playground](https://phpstan.org/r/e3239976-6e25-4395-9f7a-c300ae52fc80)) with its "[No value type specified in iterable type](https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type)" rule.

Do you think this is something that could be added to psalm? Thank you for your help!

**Comment by weirdan** [comment]
As a workaround you can use https://github.com/slevomat/coding-standard/blob/master/doc/type-hints.md#slevomatcodingstandardtypehintsparametertypehint- and https://github.com/slevomat/coding-standard/blob/master/doc/type-hints.md#slevomatcodingstandardtypehintsreturntypehint-
